esp/tools/soapplus/httpproxy.cpp

```diff
@@ -674,25 +672,28 @@ class CSocksProxyThread : public Thread
                 return -1;
             }
 
-            len += lenread;
-            m_client->read(inbuf + len, 0, 1, lenread);
+            char inbuf2[16];
+            m_client->read(inbuf2, 0, 16, lenread);
             StringBuffer username;
             while(lenread > 0)
             {
-                len += lenread;
-                if(len >= 1023)
+                bool done = false;
+                for (int i = lenread - 1; i >= 0; i--)
                 {
-                    len = 0;
+                    if (inbuf2[i] == '\0')
+                    {
+                        if (i > 0)
+                            username.append(i, inbuf2);
+                        done = true;
+                        break;
+                    }
                 }
-                if(inbuf[len - 1] == '\0')
-                {
+                if (done)
                     break;
-                }
-                char c = inbuf[len - 1];
-                username.append(c);
-                m_client->read(inbuf + len, 0, 1, lenread);
+                username.append(lenread, inbuf2);
+                m_client->read(inbuf2, 0, 16, lenread);
             }
-            
+
```

Comment:
This loop is to read until you get a null byte and then you know you are finished ?

Reply:
Yes.

Reply:
should there be a check that the username is not empty ?

Reply:
ok, a 1 byte null char empty username is allowed